app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,19 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user&.authenticate(params[:session][:password])
```

Comment:
&. để làm gì e ?

Reply:
&. 
- giúp code ngắn gọn hơn
- cách viết an toàn để đảm bảo không có các exception không mong muốn bắn ra khi object bị nil.